astropy/io/misc/parquet.py

```diff
@@ -513,3 +513,149 @@ def get_pyarrow():
         writer_version = "2.0"
 
     return pa, parquet, writer_version
+
+
+def write_parquet_votable(table, output, *, metadata, overwrite=False):
+    """
+    Writes a Parquet file with a VOT (XML) metadata table included.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        Data table that is to be written to output.
+    output : str or path-like
+        The filename to write the table to.
+    metadata : dict
+        Nested dictionary (keys = column names; sub-keys = meta keys) for each
+        of the columns containing a dictionary with metadata.
+
+    """
+    import io
+    import xml.etree.ElementTree
+
+    import pyarrow.parquet
+
+    from astropy.io.votable.tree import VOTableFile
+
+    if os.path.exists(output):
+        if overwrite:
+            # We must remove the file prior to writing below.
+            os.remove(output)
+        else:
+            raise OSError(NOT_OVERWRITING_MSG.format(output))
```

Comment:
These are explicitly tested. Would you think @pllim that the coverage test could be moved to a job that picks up all the dependencies?

Reply:
@jdavies-st once mentioned that we could run coverage on multiple jobs and aggregate the results. So technically we can add `-cov` to another job, but it does increase the run time. PR is welcome!

Reply:
I don't suggest to do it on multiple jobs, but to switch it around as it's practically useless if not misleading for e.g. this module.

Reply:
Actually, there is a cov + alldeps combo here and it has been there for at least a few years.

https://github.com/astropy/astropy/blob/3187e618bdf68a92279ad564b70dbb38de9c464b/.github/workflows/ci_workflows.yml#L79-L82

Reply:
https://github.com/astropy/astropy/blob/3187e618bdf68a92279ad564b70dbb38de9c464b/tox.ini#L109-L110

Reply:
Ahh. The job is all green but the codecov upload errors out with a rate limit message. How annoying, it should have turned the job to red to avoid sending us to wild goose chases. 

https://github.com/astropy/astropy/actions/runs/8930684539/job/24531288085#step:13:80

```
error - 2024-05-02 21:38:17,642 -- Upload failed: {"detail":"Tokenless has reached GitHub rate limit. Please upload using a token: https://docs.codecov.com/docs/adding-the-codecov-token. Expected available in 102 seconds."}
```

Reply:
Yeah, unfortunately, the rate limiting is beyond our control here. We can try to rerun it when traffic is lower.

Reply:
oh, so this is a fluke? I'll restart the job then.

But: changing the status of the job upon upload failure, or doing a sleep for 10mins and retry in the job would be things to consider.

Reply:
codecov vendors the coverage upload stuff, so I think auto retry would be a feature request upstream?